File: Final_Checkpoint/Models/Keep.cs
namespace Final_Checkpoint.Models;

public class Keep : RepoItem<int>
{
  public string Name { get; set; }
  public string Description { get; set; }
  public string Img { get; set; }
  public int Views { get; set; }
  public int Kept { get; set; }
  public string CreatorId { get; set; }
  public Profile Creator { get; set; }
}

File: Final_Checkpoint.client/src/models/Keep.js
import { Profile } from "./Account.js"
import { RepoItem } from "./RepoItem.js"

export class Keep extends RepoItem {
  constructor(data) {
    super(data)
    this.name = data.name
    this.description = data.description
    this.img = data.img
    this.kept = data.kept
    this.views = data.views
    this.creatorId = data.creatorId
    this.creator = new Profile(data.creator)
  }
}